thecourseforum/theCourseForum2
# Improve feedback

## Status

- Done

## GitHub Issues addressed

- This PR closes #566 

## What I did
- Added automated emailing to feedback/bug reports (will send a confirmation email to user from support@thecourseforum.com)
- Automatically fill in user email on bug form if logged in
- Added common send_feedback.js file with endpoints for sending discord/email messages
- Moved message formatting to backend to prevent spam messages
- Fixed form validation and cleaned up data processing

## Screenshots

- Before
-n/a
- After
Confirmation messages after submitting feedback form:
![image](https://user-images.githubusercontent.com/34071587/153048520-e8594516-90be-4a89-a13b-3e8435f27fd6.png)
![image](https://user-images.githubusercontent.com/34071587/153048644-be81a1cc-d252-41ef-88b3-49511918ecec.png)


## Tests
- Submit bug or feedback form with valid email
- Verify email confirmation message received with correct entries
- Verify discord message sent to correct channel
- Repeat with other form
- Repeat with email left blank, verify discord message sent but no email received

## Questions/Discussions/Notes
- n/a

## Merging the PR

- Who should merge this PR?
  - [] I will merge it myself
- Should the commit history be preserved in the base branch, or is it okay to combine all of them into a single commit ("squash-merge")?
  - [] preserve the history

## Add your changes to "Next Update" on the [release history page](https://github.com/thecourseforum/theCourseForum2/wiki/Release-History) once your PR gets merged!
